HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h

```diff
@@ -21,19 +23,22 @@ namespace cms {
                                               const char* cmd,
                                               const char* error,
                                               const char* message,
-                                              const char* description = nullptr) {
+                                              std::string_view description = std::string_view()) {
```

Comment:
@fwyzard Do you see any potential problems in using `std::string_view` here? (all relevant compilers for CUDA should support C++17 by some time already, right?)

Reply:
Sorry for the delay, the TDR has kept me fully busy the last few days (...).

My main concern was what happens in the vast majority of the cases, when no `description` is passed. Both @makortel and I have made some checks on godbolt, and it looks like the compiler should optimise the `std::string_view` away in that case.

As for C++17 vs earlier versions of the standard: yes, CUDA 11 fully supports C++ 17, so no problem there either.